### PR TITLE
chore: clean-up dependabot, ci, codeowners

### PR DESCRIPTION
**Pull request by joshblack**
#### New

- Add `.nvmrc` file for Node.js support

#### Changed

- Update dependabot to scan for all dependencies, include github-actions as well
- Update CI to only run on Node.js LTS (v18)
- Move `CODEOWNERS` to `.github/CODEOWNERS`

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -1,10 +1,16 @@
 version: 2
 updates:
+  - package-ecosystem: 'github-actions'
```
Comment (broccolinisoup) [inline .github/dependabot.yml:3]: I just love these 🔥

File: .github/dependabot.yml [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
+      interval: 'weekly'
+    labels:
+      - 'dependencies'
+      - 'skip changeset'
   - package-ecosystem: npm
-    directory: "/"
+    directory: '/'
     schedule:
       interval: weekly
-    open-pull-requests-limit: 99
```
Comment (broccolinisoup) [inline .github/dependabot.yml:7]: > `open-pull-requests-limit: 99`

Oo never seen this before 😄

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -1,28 +1,19 @@
...
-    - uses: actions/checkout@v2
-    - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v2
-      with:
-        node-version: ${{ matrix.node-version }}
-        cache: 'npm'
-    - run: npm ci
-    - run: npm run build --if-present
-    - run: npm test
+      - uses: actions/checkout@v2
```
Comment (broccolinisoup) [inline .github/workflows/ci.yml:12]: Seems like the last version is 4, should we update? https://github.com/actions/checkout

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -1,28 +1,19 @@
...
+      - name: Use Node.js
+        uses: actions/setup-node@v2
```
Comment (broccolinisoup) [inline .github/workflows/ci.yml:14]: Same comment as above - latest version seems to be 3 https://github.com/actions/setup-node

**Review by broccolinisoup — COMMENTED** [review]
Thanks so much for pushing these PRs! Left comments about the version of some actions - let me know what you think!

File: .github/workflows/ci.yml (hunk 3 above)
Comment (joshblack) [reply .github/workflows/ci.yml]: Done! ✨

File: .github/workflows/ci.yml (hunk 4 above)
Comment (joshblack) [reply .github/workflows/ci.yml]: Done! ✨

**Review by broccolinisoup — APPROVED** [review]

**Pull request merged**